parquet-column/src/test/java/org/apache/parquet/parser/TestParquetParser.java

```diff
@@ -310,4 +310,26 @@ public void testEmbeddedAnnotations() {
     MessageType reparsed = MessageTypeParser.parseMessageType(parsed.toString());
     assertEquals(expected, reparsed);
   }
+
+  @Test
+  public void testQuotedColumnNames() {
+    String message = "message IntMessage {" +
+        "  required int32 `foo bar`;" +
+        "  required int32 `foo.bar`;" +
+        "  required int32 `foo``bar`;" +
+        "  required int64 `foo@#$bar`;" +
+        "}\n";
+
+    MessageType parsed = MessageTypeParser.parseMessageType(message);
+    MessageType expected = Types.buildMessage()
+        .required(INT32).as(INT_8).named("foo bar")
+        .required(INT32).as(INT_16).named("foo.bar")
+        .required(INT32).as(INT_32).named("foo`bar")
+        .required(INT64).as(INT_64).named("foo@#$bar")
+        .named("IntMessage");
+
+    assertEquals(expected, parsed);
+    MessageType reparsed = MessageTypeParser.parseMessageType(parsed.toString());
+    assertEquals(expected, reparsed);
+  }
```

Comment:
we should add checks that column names that worked before do not get quoted when generating the schema now.
Otherwise older versions of parquet won't be able to read those files.